recipe/meta.yaml

```diff
@@ -690,22 +695,27 @@ outputs:
         - {{ pin_subpackage("libarrow-gandiva", exact=True) }}
         - {{ pin_subpackage("libarrow-substrait", exact=True) }}
         - {{ pin_subpackage("libparquet", exact=True) }}
```

Comment:
Why don't we just keep building against `libarrow-all` here (as before)?

Reply:
I am not sure what you mean with the `as before` we are already building against all of them because we had some issues with the `run-exports` were we had to define all of the,

Reply:
Before this PR, `pyarrow` had a host-dependency on `libarrow-all`:
https://github.com/conda-forge/arrow-cpp-feedstock/blob/a2561f808c8c794139272855ff46ba4ddbdb1e87/recipe/meta.yaml#L673-L674

The comment explicitly says that we still want all the host-dependencies, while not having all the runtime deps (for `pyarrow-core`). So my question was: why don't we just simply keep host-depending on 
```
- {{ pin_subpackage("libarrow-all", exact=True) }}
```
for `pyarrow-core`?

Reply:
I was confused. I think this was a leftover of some debugging I did. Thanks!